milestones/1-08-01-2021/tx-indexer/milestone.md

```diff
@@ -0,0 +1,168 @@
+### Milestone Title: Tx Indexer 
+### Leader  
+Eduardo Diaz Blockchain Engineer
+### Date(s)  
+01/14/2021
+### Category  
+- [ ] Infrastructure  
+- [X] Protocol/Blockchain Dev  
+- [ ] App Solutions  
+- [ ] Quality Assurance  
+### Contract  
+- [X] This process follows the Milestone Framework as discussed in the Pocket Network Corporation's `Organization Handbook`  
+- [ ] This process `DOES NOT` follow the Milestone Framework as discussed in the Pocket Network Corporation's `Organization Handbook`  
+  
+If `DOES NOT` follow the Handbook's Milestone Framework, please describe in detail why below:  
+## Original Proposal  
+[Tx indexer Proposal](https://github.com/pokt-network/organization/blob/main/proposals/1-08-01-2021/tx-indexer-milestone-proposal.md)
+
+## Milestone Detailed Summary
+The current tx indexer implemented by Tendermint team offers subpar performance by forcing us to load all index keys sort them and then build a page.
+
+this default tx indexer has turned out to be inneficcient due to: 
+- Missindexing of txs: block events get appended to all txs
+- Inneficient retrieval of data: originally tendermint loaded all tx's in memory which was causing memory overflows, we proceeded to change that however due to limitations on the implementation we still must iterate through all tx's within range sort them and then build the page.
+
+The tx indexer should be able to understand the indexes it's going through and be able to build a page without having to load all keys and sorting them.
+
+I propose we keep this format in order to keep compatibility with indexing (following format will be broken into a regular expression)
+
+```regex
+([a-z .]+)\/([a-z 0-9]+)\/([0-9]+)\/([0-9]+)[\/]?
+```
+this makes sure we get any format like this
+```
+send.action/<address>/<height>/<index>
+claim.action/<address>/<height>/<index>
+tx.block/<height>/<height>/<index>
+```
+
+This index key would allow us to:
+- easily grasp the context of the key in a humanly readable fashion
+- break down height & index to allow our store to iterate through them in order.
+
+### Milestone Goals
+- Establish a proper indexing format which should be future proof.
+- Reduce the bottleneck caused by the tendermint's implementatin. 
+
+### Proposal Estimated Timeline  
+1 Month
+
+### How to test functionality?  
+Protocol should work seamlesss, it must mantain a node should be able to query through transactions.
+
+if events are changed for this a node must resync in order to reindex all transactions.
```

Comment:
Can't we create as part of the goals of the project a CLI command that reads the transactions from the block-data and re-indexes them in this new database?

Reply:
A CLI tool can be implemented however it would require an event table in order to be able to migrate legacy indexes to new indexes. This table in turn would have to be updated with every event change performed.